### PR TITLE
build: Enable CS4014 as error

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Media_Animation/DoubleAnimationTestsControl.xaml.cs [hunk 1]
```diff
@@ -213,8 +213,7 @@ void handleSelection(object sender, object args)
 
 		private Action Dispatch(Func<CancellationToken, Task> action) => () =>
 		{
-#pragma warning disable CS4014 // Async call not awaited : Safe here (try/catch)
-			Dispatcher.RunAsync(CoreDispatcherPriority.Normal, async () =>
+			_ = Dispatcher.RunAsync(CoreDispatcherPriority.Normal, async () =>
```
Comment (Youssef1313) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Media_Animation/DoubleAnimationTestsControl.xaml.cs:216]: Looks like assigning the returned `IAsyncAction` to a discard makes the warning go away.

FYI @MartinZikmund @jeromelaban

File: src/Uno.UI.Tests/Windows_UI_Xaml_Data/xBindTests/Given_xBind_Binding.cs [hunk 2]
```diff
@@ -1088,8 +1088,8 @@ public async Task When_xLoad_StaticResource()
 
 			SUT.IsTestGridLoaded = false;
 
-			AssertIsNullAsync(() => SUT.TestGrid);
```
Comment (Youssef1313) [inline src/Uno.UI.Tests/Windows_UI_Xaml_Data/xBindTests/Given_xBind_Binding.cs:1091]: This assert is now failing after adding the missing `await` :)

Comment (Youssef1313) [reply src/Uno.UI.Tests/Windows_UI_Xaml_Data/xBindTests/Given_xBind_Binding.cs:1091]: @jeromelaban Could you take a look at this one please?

Comment (Youssef1313) [reply src/Uno.UI.Tests/Windows_UI_Xaml_Data/xBindTests/Given_xBind_Binding.cs:1091]: Note: This doesn't seem like the only test failure related to missing `await`s

Comment (jeromelaban) [reply src/Uno.UI.Tests/Windows_UI_Xaml_Data/xBindTests/Given_xBind_Binding.cs:1091]: This looks like a legitimate failure, but I don't know what it is. Let's ignore it with a new issue, I'll take a look, it's not supposed to happen.

**Review by jeromelaban — APPROVED** [review]

**Comment by Youssef1313** [comment]
@jeromelaban This is green now.

File: src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/Repeater/InspectingDataSourceTests.cs [hunk 3]
```diff
@@ -207,15 +207,15 @@ public async Task VerifyCallingResetMultipleTimesOnUniqueIdItemsSource()
...
+		await ValidateSwitchingItemsSourceRefreshesElements(isVirtualLayout: false);
 	}
 
 	[TestMethod]
-	public void ValidateSwitchingItemsSourceRefreshesElementsVirtualLayout()
+	public async Task ValidateSwitchingItemsSourceRefreshesElementsVirtualLayout()
 	{
-		ValidateSwitchingItemsSourceRefreshesElements(isVirtualLayout: true);
+		await ValidateSwitchingItemsSourceRefreshesElements(isVirtualLayout: true);
 	}
```
Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/Repeater/InspectingDataSourceTests.cs:219]: These are failing too. 😕

**Comment by jeromelaban** [comment]
Let's keep it as draft until the build passes :)

File: src/Uno.UWP/Extensions/CoreDispatcherExtensions.cs [hunk 4]
```diff
@@ -20,7 +20,7 @@ internal static async Task<T> RunWithResultAsync<T>(this CoreDispatcher dispatch
 		{
 			var tcs = new TaskCompletionSource<T>();
 
-			dispatcher.RunAsync(
+			await dispatcher.RunAsync(
```
Comment (Youssef1313) [inline src/Uno.UWP/Extensions/CoreDispatcherExtensions.cs:23]: @jeromelaban Should this assign to a discard? I don't know if it should be fire-and-forget or not.

**Comment by Youssef1313** [comment]
@jeromelaban Tests timed out. I restarted them, but if they timed out again then I don't understand why or if that's related to any of the changes. Hence, I'm marking as ready for review.

**Comment by jeromelaban** [comment]
It may be related because tests are now taking longer to run because of the awaits ?

**Review by carldebilly — APPROVED** [review]

File: src/Uno.UWP/Extensions/CoreDispatcherExtensions.cs (hunk 4 above)
Comment (jeromelaban) [reply src/Uno.UWP/Extensions/CoreDispatcherExtensions.cs:23]: Let's keep it that way. It's not used much, and not in critical paths so we can validate with the additional await.

**Pull request merged**